logzio-sender/src/main/java/io/logz/sender/HttpsSyncSender.java

```diff
@@ -63,52 +51,30 @@ private int sizeInBytes(List<FormattedLogMessage> logMessages) {
         return totalSize;
     }
 
-    public void sendToLogzio(List<FormattedLogMessage> messages) throws LogzioServerErrorException {
+    void sendToLogzio(List<FormattedLogMessage> messages) throws LogzioServerErrorException {
         try {
             byte[] payload = toNewLineSeparatedByteArray(messages);
             int currentRetrySleep = configuration.getInitialWaitBeforeRetryMS();
 
             for (int currTry = 1; currTry <= configuration.getMaxRetriesAttempts(); currTry++) {
-                boolean shouldRetry = true;
+                boolean retry = true;
                 int responseCode = 0;
                 String responseMessage = "";
                 IOException savedException = null;
 
                 try {
-                    HttpURLConnection conn = getHttpURLConnection(payload);
+                    HttpURLConnection conn = sendRequest(payload);
                     responseCode = conn.getResponseCode();
                     responseMessage = conn.getResponseMessage();
-
-                    if (responseCode == HttpURLConnection.HTTP_BAD_REQUEST) {
-                        readBadRequest(conn);
-                    }
-                    if (responseCode == HttpURLConnection.HTTP_UNAUTHORIZED) {
-                        reporter.error("Logz.io: Got forbidden! Your token is not right. Unfortunately, dropping logs. Message: " + responseMessage);
-                    }
-
-                    shouldRetry = shouldRetry(responseCode);
+                    retry = shouldRetry(payload, responseCode, responseMessage, conn);
```

Comment:
Maybe `shouldRetry = handleResponse(...)`? You're doing a lot more that deciding if you should retry